docs/Open Data Model/Open Data Model.md

```diff
@@ -0,0 +1,755 @@
+For example, if a user wanted to extend the user context model to include a string attribute for “Desk Location” and “City”, the following string would be set for additional_attrs:
+
+|**Attribute Key**|**Attribute Value**|
+|---|---|
+|additional_attrs|{"dsk_location":"B3-F2-W3", "city":"Palo Alto"}|
+
+
+Something similar can be done for endpoint context model, security event log/alert model and other entities.
+
+	Note: This [UDF library](https://github.com/klout/brickhouse) can be used for converting to/from JSON.
+
+##**Model Relationships**##
+
+The relationships between the data model entities are illustrated below.
+
+Image here
+
+
+##**Data Ingestion Framework**##
+
+One of the challenges in populating the data model is the large number of products and technologies that organizations are currently using to manage security event logs/alerts, user and endpoint information. There are literally dozens of vendors in each category that offer technologies that could be used to populate the model.  The labor required to transform the data and map the attributes to the data model is extensive when you consider how many technologies are in the mix at each organization (and across organizations). One way to address this challenge is with a Data Ingestion Framework that provides a configuration-based mechanism to perform the transformations and mappings.  A configuration-based capability will allow the ingest pipelines to become portable and reusable across the community. For example, if I create an ingest pipeline for Centrify to populate the user context model, it can be shared with other users of Centrify who can immediately realize the benefit.  Such a framework could allow the community to quickly build the necessary pipelines for the dozens (and hundreds) of technologies being used in the market.  Without a standard ingest framework, each pipeline is built independently, requiring more labor, providing no standardization and little portability.  It’s also important that the data ingestion framework support the ability to both capture the “raw” event and create a meta event that represents the normalized event and maps the attributes to the defined data model.  This will ensure both stream and batch processing use cases are supported.
+ 
+Streamsets is an ingest framework that provides the needed functionality outlined above.  Sample Streamsets ingest pipelines for populating the ODM with common data sources will be published to the Spot Github repo.
+
+##**Data Formats**##
+
+**Avro**
+----
+
+Avro is the recommended data format due to its schema representation, compatibility checks, and interoperability with Hadoop.  Avro supports a pure JSON representation for readability and ease of use but also a binary representation of the data for efficient storage.   Avro is the optimal format for streaming-based analytic use cases.
+
+A sample event and corresponding schema representation are detailed below.
+
+**Event**
```

Comment:
This may get compacted in one line so you have to put 3 backticks at top and bottom of this block. Same for Schema.